### PR TITLE
Move quarkus-devtools-testing to the quarkus-bom as it's used for extension dev

### PR DESCRIPTION
**Pull request by aloubyansky**
Specifically codestart dev and testing.

### CODE REVIEW

**Review by ia3andy — APPROVED** [review]
LGTM smart move

**Comment by ia3andy** [comment]
@aloubyansky what about `devtools-common`? did you see it's a dependency of devtools-testing?

**Comment by gsmet** [comment]
I'm not sure that's related but having both builds failing at the exact same step is suspicious so I will not backport for now.

File: bom/application/pom.xml [hunk 1]
```diff
@@ -5608,6 +5608,17 @@
                 <artifactId>quarkus-devtools-registry-client</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-devtools-testing</artifactId>
+                <scope>test</scope>
```
Comment (gsmet) [inline bom/application/pom.xml:5614]: We usually don't define the scope in the BOM as it's bad practice.

**Comment by gsmet** [comment]
I see the failures in other PRs so it doesn't seem related. I'm not sure about the scope thing though, I'd prefer we clear this up before merging/backporting.

File: bom/application/pom.xml (hunk 1 above)
Comment (aloubyansky) [reply bom/application/pom.xml:5614]: Ok, I'll remove it.

**Comment by gsmet** [comment]
I pushed it to 2.5 and restarted the release process.

**Comment by kshpak** [comment]
This PR broke CLI build
`cd devtools/cli && mvn -B -s ../../.github/mvn-settings.xml clean install -Dquickly ` Throws:

```
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for io.quarkus:quarkus-devtools-common:jar is missing. @ line 25, column 21
[ERROR] 'dependencies.dependency.version' for io.quarkus:quarkus-devtools-testing:jar is missing. @ line 61, column 21
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project io.quarkus:quarkus-cli:999-SNAPSHOT (/home/kshpak/myTests/Quarkus/github/quarkus-main/devtools/cli/pom.xml) has 2 errors
[ERROR]     'dependencies.dependency.version' for io.quarkus:quarkus-devtools-common:jar is missing. @ line 25, column 21
[ERROR]     'dependencies.dependency.version' for io.quarkus:quarkus-devtools-testing:jar is missing. @ line 61, column 21
```

**Comment by aloubyansky** [comment]
@kshpak have you re-built the branch?

**Pull request merged**